extensions/amp-ad-network-adsense-impl/0.1/amp-ad-network-adsense-impl.js

```diff
@@ -151,47 +153,23 @@ export class AmpAdNetworkAdsenseImpl extends AmpA4A {
   emitLifecycleEvent(eventName, opt_associatedEventData) {
     this.lifecycleReporter_ = this.lifecycleReporter_ ||
```

Comment:
Oops.  We actually do.  We're currently calling `emitLifecycleEvent` in the superclass constructor.  But the subclass hasn't initialized it yet.  So `emitLifecycleEvent` needs to be prepared to, if necessary.  I tweaked the way it worked, though, so that we're not testing existence on every call.  The new version is a bit...  JavaScript-y.  :-)  Take a look and see if it's too fugly.